File: gitdot-api/src/endpoint/repository/star_repository.rs
use serde::{Deserialize, Serialize};

use crate::endpoint::Endpoint;

pub struct StarRepository;

impl Endpoint for StarRepository {
    const PATH: &'static str = "/repository/{owner}/{repo}/star";
    const METHOD: http::Method = http::Method::POST;

    type Request = StarRepositoryRequest;
    type Response = StarRepositoryResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct StarRepositoryRequest {}

pub type StarRepositoryResponse = ();

File: gitdot-web/packages/gitdot-api/src/endpoint/repository/star-repository.ts
import { z } from "zod";
import type { Endpoint } from "../endpoint";

export const StarRepositoryRequest = z.object({});
export type StarRepositoryRequest = z.infer<typeof StarRepositoryRequest>;

export const StarRepositoryResponse = z.void();
export type StarRepositoryResponse = z.infer<typeof StarRepositoryResponse>;

export const StarRepository = {
  path: "/repository/{owner}/{repo}/star",
  method: "POST",
  request: StarRepositoryRequest,
  response: StarRepositoryResponse,
} as const satisfies Endpoint;
export type StarRepository = typeof StarRepository;
